### PR TITLE
Only push current branch on publish

### LINKED ISSUE #88: Git push only current branch
We're currently pushing every branch on lerna publish which breaks the publishing process if you have branches that are rejected.

```
Publishing tagged packages...
Setting latest npm tags...
Removing temporary npm dist-tag
Adding latest npm dist-tag
npm WARN dist-tag add latest is already set to version 1.0.0
To ssh://git@xxxx.git
   aafeff6..f74cc37  master -> master
 ! [rejected]        form -> form (non-fast-forward)
```

### PR DESCRIPTION
**Pull request by thejameskyle**
Fixes #88

### CODE REVIEW

File: test/PublishCommand.js [hunk 1]
```diff
@@ -71,7 +71,8 @@ describe("PublishCommand", () => {
           { args: ["npm dist-tag rm package-4 lerna-temp"] },
           { args: ["npm dist-tag add package-4@1.0.1 latest"] },
 
-          { args: ["git push"] },
+          { args: ["git symbolic-ref --short HEAD"], returns: "master", returns: "master" },
```
Comment (hzoo) [inline test/PublishCommand.js]: I see - there's 2 `returns: "master"` and fails lint

File: src/GitUtilities.js [hunk 2]
```diff
@@ -40,7 +40,7 @@ export default class GitUtilities {
 
   @logger.logifySync
   static pushWithTags(tags) {
-    ChildProcessUtilities.execSync("git push");
+    ChildProcessUtilities.execSync("git push origin " + GitUtilities.getCurrentBranch());
```
Comment (hzoo) [inline src/GitUtilities.js]: Not a bad assumption, but we assume the branch is always `origin` in these 2 lines - I guess that could go in config (https://github.com/kittens/lerna/issues/42)?

File: test/PublishCommand.js (hunk 1 above)
Comment (thejameskyle) [inline test/PublishCommand.js]: :expressionless:

File: test/PublishCommand.js [hunk 3]
```diff
@@ -165,7 +166,8 @@ describe("PublishCommand", () => {
           { args: ["npm dist-tag rm package-4 lerna-temp"] },
           { args: ["npm dist-tag add package-4@1.1.0 latest"] },
 
-          { args: ["git push"] },
+          { args: ["git symbolic-ref --short HEAD"], returns: "master", returns: "master" },
```
Comment (hzoo) [inline test/PublishCommand.js]: sorry 😢  same here

**Comment by hzoo** [comment]
👍

File: src/GitUtilities.js (hunk 2 above)
Comment (thejameskyle) [inline src/GitUtilities.js]: Yeah, we could attempt to lookup what the tracked remote branch is:

```
$ repo git:(branch): git for-each-ref --format='%(upstream:short)' $(git symbolic-ref -q HEAD)
> origin/master
```

I don't know how reliable that would be though

Comment (hzoo) [inline src/GitUtilities.js]: I think it's cool we can just note it in the help command, and the readme

**Pull request merged**